File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/entity/EntityExposureSummary.java
package org.coralibre.android.sdk.internal.database.persistent.entity;


import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

import org.coralibre.android.sdk.fakegms.nearby.exposurenotification.ExposureSummary;
import org.coralibre.android.sdk.internal.datatypes.AssociatedEncryptedMetadata;
import org.coralibre.android.sdk.internal.datatypes.CapturedData;
import org.coralibre.android.sdk.internal.datatypes.ENInterval;
import org.coralibre.android.sdk.internal.datatypes.RollingProximityIdentifier;

import static androidx.room.ForeignKey.CASCADE;


@Entity(foreignKeys = @ForeignKey(
    entity = EntityToken.class,
    parentColumns = "tokenString",
    childColumns = "tokenString",
    onDelete = CASCADE
))
public class EntityExposureSummary {

    /**
     * This field is not used. It is only here to give the database a primary key. When accessing
     * data from the database, we usually filter by timestamp.
     */
    @PrimaryKey(autoGenerate = false)
    @NonNull
    public String tokenString;

    public int daysSinceLastExposure;
    public int matchedKeyCount;
    public int maximumRiskScore;
    public int summedAttenuationDurationBelow;
    public int summedAttenuationDurationBetween;
    public int summedAttenuationDurationAbove;
    public int summationRiskScore;


    public EntityExposureSummary(){}

    public EntityExposureSummary(String token, ExposureSummary summary) {
        daysSinceLastExposure = summary.getDaysSinceLastExposure();
        matchedKeyCount = summary.getMatchedKeyCount();
        maximumRiskScore = summary.getMaximumRiskScore();
        summedAttenuationDurationBelow = summary.getAttenuationDurations()[0];
        summedAttenuationDurationBetween = summary.getAttenuationDurations()[1];
        summedAttenuationDurationAbove = summary.getAttenuationDurations()[2];
        summationRiskScore = summary.getSummationRiskScore();
        tokenString = token;
    }

    public ExposureSummary toExposureSummary() {
        int[] attenuationDurations = {
            summedAttenuationDurationBelow,
            summedAttenuationDurationBetween,
            summedAttenuationDurationAbove
        };
        return new ExposureSummary(
            daysSinceLastExposure,
            matchedKeyCount,
            maximumRiskScore,
            attenuationDurations,
            summationRiskScore
        );
    }

}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/entity/EntityExposureSummary.kt
package org.coralibre.android.sdk.internal.database.persistent.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import org.coralibre.android.sdk.fakegms.nearby.exposurenotification.ExposureSummary

/**
 * @param tokenString  This field is not used. It is only here to give the database a primary key.
 * When accessing data from the database, we usually filter by timestamp.
 */
@Entity(
    foreignKeys = [
        ForeignKey(
            entity = EntityToken::class,
            parentColumns = ["tokenString"],
            childColumns = ["tokenString"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class EntityExposureSummary(
    @PrimaryKey(autoGenerate = false)
    val tokenString: String,
    val daysSinceLastExposure: Int,
    val matchedKeyCount: Int,
    val maximumRiskScore: Int,
    val summedAttenuationDurationBelow: Int,
    val summedAttenuationDurationBetween: Int,
    val summedAttenuationDurationAbove: Int,
    val summationRiskScore: Int,
) {
    constructor(token: String, summary: ExposureSummary) : this(
        daysSinceLastExposure = summary.daysSinceLastExposure,
        matchedKeyCount = summary.matchedKeyCount,
        maximumRiskScore = summary.maximumRiskScore,
        summedAttenuationDurationBelow = summary.attenuationDurations[0],
        summedAttenuationDurationBetween = summary.attenuationDurations[1],
        summedAttenuationDurationAbove = summary.attenuationDurations[2],
        summationRiskScore = summary.summationRiskScore,
        tokenString = token,
    )

    fun toExposureSummary(): ExposureSummary {
        val attenuationDurations = intArrayOf(
            summedAttenuationDurationBelow,
            summedAttenuationDurationBetween,
            summedAttenuationDurationAbove
        )
        return ExposureSummary(
            daysSinceLastExposure,
            matchedKeyCount,
            maximumRiskScore,
            attenuationDurations,
            summationRiskScore
        )
    }
}
